shared-bindings/wifi/Monitor.c

```diff
@@ -0,0 +1,165 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2021 microDev
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "py/mpstate.h"
+#include "py/runtime.h"
+#include "py/objproperty.h"
+
+#include "shared-bindings/util.h"
+#include "shared-bindings/wifi/Packet.h"
+#include "shared-bindings/wifi/Monitor.h"
+
+//| class Monitor:
+//|     """For monitoring WiFi packets."""
+//|
+
+//| def __init__(self, channel: Optional[int] = 1, queue: Optional[int] = 128) -> None:
+//|     """Initialize `wifi.Monitor` singleton.
+//|
+//|     :param int channel: The WiFi channel to scan.
+//|     :param int queue: The queue size for buffering the packet.
+//|
+//|     """
+//|     ...
+//|
+STATIC mp_obj_t wifi_monitor_make_new(const mp_obj_type_t *type, size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+    enum { ARG_channel, ARG_queue };
+    static const mp_arg_t allowed_args[] = {
+        { MP_QSTR_channel, MP_ARG_INT | MP_ARG_KW_ONLY, {.u_int = 1} },
+        { MP_QSTR_queue, MP_ARG_INT | MP_ARG_KW_ONLY, {.u_int = 128} },
+    };
+
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args, pos_args, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    if (args[ARG_channel].u_int < 0 || args[ARG_channel].u_int > 11) {
```

Comment:
Channels 12 and 13 are valid in various areas outside the U.S. For whatever reason, I am also seeing packets on channels 12 and 13 at my U.S. location (`monitor.channel` does not have a limit check). I suggest the limit check allow channels 1-13.

Reply:
Increased the limit. :)